### PR TITLE
Default ssl configuration within security Provider

### PR DESCRIPTION
**Pull request by Hakky54**

### CODE REVIEW

**Comment by Hakky54** [comment]
Hi @balbusm 

I created an initial implementation for the feature request. With this current implementation your use case would be:

```java
X509ExtendedTrustManager trustManager = PemUtils.loadTrustMaterial(Paths.get(truststorePath));
SSLFactory sslFactory  = SSLFactory.builder()
                .withDefaultTrustMaterial()
                .withTrustMaterial(trustManager)
                .build();

SSLFactoryUtils.configureAsDefault(sslFactory);
```

I have added the code snippet of configuring the security provider in the SSLFactoryUtils or else it would be in two places, KeyManagerUtils and TrustManager. 

What do you think of the solution and the usage of it, can you maybe have a look at it and share your thoughts about it, maybe some feedback?

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/SSLFactoryUtils.java [hunk 1]
```diff
@@ -50,4 +55,20 @@ private static <T> void reload(SSLFactory baseSslFactory,
         }
     }
 
+    public static void configureAsDefault(SSLFactory sslFactory) {
```
Comment (balbusm) [inline sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/SSLFactoryUtils.java:58]: Method serves original request from the issue #250, but it's misleading.
Looking at location (SSLFactoryUtils) and name I would assume it sets entire SSLContext as default.
I see 2 options:
- move it to TrustManagerUtils
- Create RootSSLContextSpi similar to RootTrustManagerSpi

Comment (balbusm) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/SSLFactoryUtils.java:58]: There is also third option change the name of the method

Comment (Hakky54) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/SSLFactoryUtils.java:58]: Sounds good, however I discovered that the RootKeymanagerFactorySpi is not working as well as the SSLContextSpi. The SSLContextSpi I have not comitted yet. Do you have working code examples for those two?

**Comment by Hakky54** [comment]
During the testing I discovered the configuration for the trustmanager works with 95% and the keymanager works with 30% of the tested http clients. I used my other repo for testing all of the http clients, which can be found here: https://github.com/Hakky54/mutual-tls-ssl#tested-clients The configuration in this PR is correct, however a-lot of http clients won't pick it up. Next to that it messes op the initialization of a new KeyManagerFactory/TrustManagerFactory when you need a new one for example when someone tries to reload the ssl configuration. I like the idea of configuring the ssl configuration globally, but sadly it does not work with all of the http clinets which is for me a no-go and next to that it will break other functionalities. Therefor I am closing the PR and issue. Thank you for creating the feature request and sharing your knowledge @balbusm !

**Pull request closed without merging**